### PR TITLE
[10.0][FIX] Subcontracted service: can't adapt type 'procurement.rule'

### PR DESCRIPTION
**Pull request by jvpascual**
when a sale order is confirmed, 
ProgrammingError: can't adapt type 'procurement.rule'

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@hveficent can you review?

**Comment by HviorForgeFlow** [comment]
@jvpascual  I cannot reproduce error you've got. Any idea about which modulo or configuration has made it raise?

**Comment by jvpascual** [comment]
Hi @hveficent,
in the attached error trace, can see the modules, the unique that is not in oca is vit_mrp_cost,
imagen

Kind regards,

**Comment by HviorForgeFlow** [comment]
@jvpascual can you copy the full log in plain text??

**Comment by jvpascual** [comment]
Hi @hveficent, the file is attached.
[subcontracted_service_error.txt](https://github.com/OCA/purchase-workflow/files/2274210/subcontracted_service_error.txt)

File: subcontracted_service/models/procurement.py [hunk 1]
```diff
@@ -31,6 +31,9 @@ def _assign(self):
         if not res:
             rule_id = self._find_suitable_rule()
             if rule_id:
-                self.write({'rule_id': rule_id})
```
Comment (HviorForgeFlow) [inline subcontracted_service/models/procurement.py:34]: change to self.write({'rule_id': rule_id.id})

Comment (HviorForgeFlow) [reply subcontracted_service/models/procurement.py:34]: in _find_suitable_rule if subcontracted service remove last .id like:
self.warehouse_id.subcontracting_service_proc_rule_id

Comment (HviorForgeFlow) [reply subcontracted_service/models/procurement.py:34]: I think this way it should work, could you try in your instance?

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]

File: subcontracted_service/models/procurement.py (hunk 1 above)
Comment (jvpascual) [reply subcontracted_service/models/procurement.py:34]: Yes, it works.
i've just leave the old self.write({'rule_id': rule_id}) in case it come as object.

**Comment by HviorForgeFlow** [comment]
Travis is failing. You must attend the other change, I think you have some module that is overwriting/extending function _find_suitable_rule(), could you find out which module it is..

**Comment by jvpascual** [comment]
the function _find_suitable_rule(), only appears in the next modules: stock, procurement and subcontracted_service.
in some cases, can arrive as an object.

**Comment by HviorForgeFlow** [comment]
Ok, I got it. But then fix _find_suitable_rule in this module to return an object and keep rule_id write as previous commit.

**Comment by jvpascual** [comment]
Done.

**Review by HviorForgeFlow — APPROVED** [review]
Please squash commits into 1

**Comment by jvpascual** [comment]
i've lost the origin.

**Pull request closed without merging**